PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -212,9 +220,15 @@ def __call__(self, process,
         self.setParameter('postfix',postfix),
 
         #if mva/puppi MET, autoswitch to std jets
-        if metType == "MVA" or 'Puppi' in postfix:
+        if metType == "MVA" or metType == "Puppi":
             self.setParameter('CHS',False),
 
+        #enabling puppi flag
+        self.setParameter('Puppi',self._defaultParameters['Puppi'].value) 
+        if metType == "Puppi":
+            self.setParameter('metType',"PF") 
+            self.setParameter('Puppi',True) 
```

Comment:
@slava77 I agree can be improved but this require some redesign from matthieu, the current solution practically is needed to fix the content